cmd/cert-checker/main.go

```diff
@@ -202,48 +190,19 @@ func (c *certChecker) checkCert(cert core.Certificate) (problems []string) {
 		problems = append(problems, "Stored digest doesn't match certificate digest")
 	}
 
-	// Run linter
-	linter := new(lintasn1.Linter)
-	errs := linter.CheckStruct(cert.DER)
-	if errs != nil {
-		for _, err := range errs.List() {
-			problems = append(problems, err.Error())
-		}
-	}
-	d, err := certdata.Load(cert.DER)
-	if err != nil {
-		problems = append(problems, err.Error())
-	}
-	errs = checks.Certificate.Check(d)
-	if errs != nil {
-		for _, err := range errs.List() {
-			// commonName has been deprecated for years, but common practice is still
-			// to include it for compatibility reasons. For instance, Chrome on macOS
-			// until very recently would error on an empty Subject (which is what we
-			// would have if we omitted CommonName). There have been proposals at
-			// CA/Browser Forum for an alternate contentless field whose purpose would
-			// just be to make Subject non-empty, but so far they have not been
-			// successful. If the check error is `certlintCNError`, ignore it.
-			if err.Error() == certlintCNError {
-				continue
-			}
-			// certlint incorrectly flags certificates that have a subj. CN or SAN
-			// exactly equal to a *private* entry on the public suffix list. Since
-			// this is allowed and LE issues certificates for such names we ignore
-			// errors of this form until the upstream bug can be addressed. See
-			// https://github.com/globalsign/certlint/issues/17
-			if certlintPSLErrPattern.MatchString(err.Error()) {
-				continue
-			}
-			problems = append(problems, err.Error())
-		}
-	}
-
 	// Parse certificate
 	parsedCert, err := x509.ParseCertificate(cert.DER)
 	if err != nil {
 		problems = append(problems, fmt.Sprintf("Couldn't parse stored certificate: %s", err))
 	} else {
+		// Run zlint checks
+		results := zlint.LintCertificate(parsedCert)
+		for name, res := range results.Results {
+			// ignore notices and warnings
+			if res.Status >= lints.Error {
+				problems = append(problems, fmt.Sprintf("zlint %s: %s", res.Status, name))
```

Comment:
Can you also include the `LintResult`'s `Details` string (maybe just if it's `!= ""`) in addition to the `Status`?